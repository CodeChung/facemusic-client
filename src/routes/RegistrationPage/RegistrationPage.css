.registration-page {
    height: 70%;
    margin: 0 auto;
    width: 40%;
}

.registration-page h2 {
    margin: 0;
}

.registration-form label {
    display: none;
}

.registration-form input {
    border: none;
    border-bottom: 1px solid gray;
    background-color: #F1FFFA;
}

.registration-form input {
    margin: 15px 0;
    width: 60%;
    font-size: 20px;
}

.registration-form input:focus {
    outline: none;
}

.registration-form button {
    width: 30%;
    margin-top: 30px;
    padding: 10px;
    border-radius: 20px;
    border: none;
    background-color: rgb(255, 255, 255);
}

.registration-form button:focus {
    outline: none;
}

.registration-form button:active {
    background-color: rgb(163, 163, 163);
}

@media(max-width:768px){
    .registration-page {
        width: 70%;
    }
}