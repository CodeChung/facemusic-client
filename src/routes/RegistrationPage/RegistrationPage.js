import React, { Component } from 'react'
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm'

export default class RegistrationPage extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  }

  handleRegistrationSuccess = user => {
    const { history } = this.props
    history.push('/login')
  }

  render() {
    return (
      <section className='registration-page'>
        <h2>Register</h2>
        <RegistrationForm
          onRegistrationSuccess={this.handleRegistrationSuccess}
        />
      </section>
    )
  }
}
