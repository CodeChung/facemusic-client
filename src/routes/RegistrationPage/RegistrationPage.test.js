import React from 'react';
import ReactDOM from 'react-dom';
import RegistrationPage from './RegistrationPage';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe(`RegistrationPage Component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RegistrationPage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('displays a valid RegistrationPage', () => {
    const wrapper = shallow(<RegistrationPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})