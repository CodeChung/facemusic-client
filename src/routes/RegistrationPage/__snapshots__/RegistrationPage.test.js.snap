// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationPage Component displays a valid RegistrationPage 1`] = `
<section
  className="registration-page"
>
  <h2>
    Register
  </h2>
  <RegistrationForm
    onRegistrationSuccess={[Function]}
  />
</section>
`;
